extensions/vertx-web/deployment/src/test/java/io/quarkus/vertx/web/validation/SyncValidationTest.java

```diff
@@ -0,0 +1,147 @@
+package io.quarkus.vertx.web.validation;
+
+import static io.restassured.RestAssured.get;
+import static io.restassured.RestAssured.given;
+import static org.hamcrest.CoreMatchers.anyOf;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.is;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.validation.constraints.Pattern;
+
+import org.hibernate.validator.constraints.Length;
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.JavaArchive;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import io.quarkus.test.QuarkusUnitTest;
+import io.quarkus.vertx.web.Param;
+import io.quarkus.vertx.web.Route;
+import io.vertx.core.http.HttpMethod;
+
+public class SyncValidationTest {
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .setArchiveProducer(() -> ShrinkWrap.create(JavaArchive.class)
+                    .addClasses(MyRoutes.class));
+
+    @Test
+    public void test() {
+        // Valid result
+        get("/valid").then().statusCode(200)
+                .body("name", is("luke"))
+                .body("welcome", is("hello"));
+
+        // Valid parameter
+        given()
+                .queryParam("name", "neo")
+                .when()
+                .get("/query")
+                .then().statusCode(200);
+
+        // JSON output
+        given()
+                .header("Accept", "application/json")
+                .when()
+                .get("/invalid")
+                .then()
+                .statusCode(500)
+                .body("title", containsString("Constraint Violation"))
+                .body("status", is(500))
+                .body("details", containsString("validation constraint violations"))
+                .body("violations[0].field", containsString("name"))
+                .body("violations[0].message", containsString("length"));
```

Comment:
These tests will fail on systems with non-EN locale...

Reply:
I think the default locale for a JVM is derived from the OS and in my case, it's `cs_CZ.UTF-8`. So I get czech messages for the constraints violations...

Reply:
Checking for nonblank.